src/plugins/intel_gpu/include/intel_gpu/primitives/shape_of.hpp

```diff
@@ -31,6 +31,17 @@ struct shape_of : public primitive_base<shape_of> {
         : primitive_base(id, {input}, output_padding, optional_data_type{output_data_type})
         , output_rank(output_rank) {}
 
+    /// @brief Constructs shape_of primitive.
+    /// @param id This primitive id.
+    /// @param input Input primitive id.
+    /// @param output_data_type type of output values. can be i32 and i64.
+    shape_of(const primitive_id& id,
+             const primitive_id& input,
+             const data_types output_data_type,
+             const padding& output_padding = padding())
+        : primitive_base(id, {input}, output_padding, optional_data_type{output_data_type})
+        , output_rank(0) {}
```

Comment:
I think we can just merge those two APIs to one with default value.

Reply:
We can merge for sure, but my idea is to separate api which can't be fixed immediately, and then just remove it once we migrate to new shape infer.